bpf/lib/common.h

```diff
@@ -664,6 +668,18 @@ enum metric_dir {
 #define DSR_IPV6_OPT_LEN	(sizeof(struct dsr_opt_v6) - 4)
 #define DSR_IPV6_EXT_LEN	((sizeof(struct dsr_opt_v6) - 8) / 8)
 
+/* Geneve option used to carry service addr and port for DSR.
+ *
+ * Class = 0x200 (unassigned according to [1])
+ * Type  = 0x20  (vendor-specific)
+ *
+ * [1]: https://www.iana.org/assignments/nvo3/nvo3.xhtml#geneve-option-class
+ */
+#define DSR_GENEVE_OPT_CLASS	0x200
+#define DSR_GENEVE_OPT_TYPE	0x20
```

Comment:
Is there any particular pattern to this option type? `0x20` seems like an odd place to start. I would have expected to start from maybe `0x1` for the first such option in this new option class (offset by the critical bit if applicable).

It's probably also worth considering whether the 0th bit in this opt type should be set, to signal that the option is `Critical` cf. https://www.rfc-editor.org/rfc/rfc8926.html#name-tunnel-options . Given that the option contains the original service address for replying to the connection, and the remote node won't have that context, I think that this option *is* critical information that the destination node cannot ignore.

Reply:
There's no particular reason for `0x20`. I just picked it up randomly. It makes sense to start from `0x1`.

> It's probably also worth considering whether the 0th bit in this opt type should be set, to signal that the option is Critical

Thank you for the suggestion! I will look into that